HW_2/app.py

```diff
@@ -0,0 +1,32 @@
+from flask import Flask, render_template, request, redirect, url_for, make_response
+
+app = Flask(__name__)
+app.secret_key = 'JohnDoe' 
+
+@app.route('/')
+def index():
+    return render_template('index.html')
+
+@app.route('/welcome', methods=['POST'])
+def welcome():
+    if request.method == 'POST':
+        username = request.form['username']
+        email = request.form['email']
+
+        response = make_response(render_template('welcome.html', username=username))
+        response.set_cookie('username', username)
+        response.set_cookie('email', email)
+
+        return response
+
+@app.route('/logout', methods=['POST'])
+def logout():
+
+    response = make_response(redirect(url_for('index')))
+    response.delete_cookie('username')
+    response.delete_cookie('email')
+
+    return response
```

Comment:
Можно совместить две функции и поменять третью
```suggestion
@app.route('/')
def index():
    return render_template('index.html')

@app.route('/', methods=['GET', 'POST'])
def index():
    if request.method == 'POST':
        username = request.form['username']
        email = request.form['email']

        response = make_response(redirect('welcome'))
        response.set_cookie('username', username)
        response.set_cookie('email', email)

        return response

    return render_template('index.html')


@app.route('/welcome', methods=['GET', 'POST'])
def welcome():
    if request.method == 'POST':
        response = make_response(redirect(url_for('index')))
        response.delete_cookie('username')
        response.delete_cookie('email')

        return response

    username = request.cookies.get('username')
    return make_response(render_template('welcome.html', username=username))
```